### DISCUSSION TITLE
How to fit everthing to screen/window width?

### DISCUSSION THREAD

**Discussion by hopezh** [quarto-dev/quarto-cli#9584] [Q&A] [answered]
### Description

I used `::: {.column-screen}` to wrap all the contents in a page,  including side bar on the left, contents in main page in the middle, and table of contents on the right, but everything still appears in the middle of the screen/window, rather than utilizing the full span of the screen/window. 

May I ask how to achieve this? 

Test repo is [here](https://github.com/hopezh/oatNotes), and the qmd of the page is [here](https://github.com/hopezh/oatNotes/blob/main/pages/rhino3dm/test_rhino3dm.qmd). 

```md
---
title: "test rhino3dm"
---

:::: {.column-screen}

(contents of the page)

::::
```

Screen image:
![2024-05-05 15 27 26](https://github.com/quarto-dev/quarto-cli/assets/23435810/3e629140-750c-4d0b-a6f6-e8586034cb77)

**Comment by mcanouil** [comment, marked as answer]
I don't understand `.column-screen` is only for main content as described in <https://quarto.org/docs/authoring/article-layout.html>.
This won't affect sidebar/margin.

<table>
<tr><th>Quarto document</th><th>HTML</th></tr>
<tr><td>

````qmd
---
format: html
---

{{< lipsum 1 >}}

::: {.column-screen}
![](https://placehold.co/1600x400.png)

{{< lipsum 1 >}}
:::
````

</td><td>

<img width="1512" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/d5acf060-d46c-4538-8cd8-7099d9b7225a">

</td></tr>
</table>


If you want to control the page layout, see <https://quarto.org/docs/output-formats/page-layout.html>

**Reply by hopezh** [reply]
Thanks.

I added `page-layout: full` in `_quarto.yml`:

```md
format:
  html:
    page-layout: full
```

However, all the contents are still presented at the center rather than spanning to the full window width as indicated by the red arrow below:

![2024-05-05 19 06 17](https://github.com/quarto-dev/quarto-cli/assets/23435810/62d2d984-c0f2-4061-8f29-d8fdb1eb4591)

So, I tried the `grid` to manually specify the width of side bar, main page, and table of contents, and it works.

```md
format:
  html:
    grid:
      sidebar-width: 300px
      body-width: 2000px
      margin-width: 300px
      gutter-width: 1.5rem
```

![2024-05-05 19 17 16](https://github.com/quarto-dev/quarto-cli/assets/23435810/057a974d-4589-463e-8d6f-29c88b4ae9cb)

**Reply by mcanouil** [reply]
Does the grid defining your website layout good enough solution?

**Reply by hopezh** [reply]
Yes, @mcanouil, the[ grid customization](https://quarto.org/docs/output-formats/page-layout.html#grid-customization) is a very flexible solution. Thanks.
